### ISSUE TITLE
YOLO v4 is not as well in the scene 

### ISSUE THREAD

**Issue by wwzh2015** [AlexeyAB/darknet#5567]
This is YOLO v4
https://imgchr.com/i/YYkmgH

This is EfficientDet D4
https://s1.ax1x.com/2020/05/11/YYkt2Q.jpg

**Comment by AlexeyAB** [comment]
1. YOLOv4 was trained on MSCOCO-2014 that is labeled with crowds=1, so the crowd of people is marked as one object in the MSCOCO-dataset, as you see in your yolov4-detection image. This can greatly reduce the detection accuracy of individual persons in a crowd. So YOLOv4 should be retrained with correct labels.

2. For each neural network, you can find a picture with which one neural network is working poorly and the other is good, and vice versa. Therefore, for testing instead of 1 image there are used 20,000 images from the Microsoft COCO test-dev dataset

3. EfficientDet D4 -  49.0% AP - **13 FPS** (Volta), while YOLOv4 - 43.5% AP - **62 FPS** (Volta), https://arxiv.org/pdf/1911.09070.pdf and https://arxiv.org/pdf/2004.10934.pdf so yes EfficientDet D4 is **+5.5% more accurate and 4.7x time slower** than YOLOv4

4. EfficientDet D4 was trained on network 1024x1024 while YOLOv4 was trained on 512x512, so EfficientDet D4 can better detect small objects

5. Did you ever try to train and test EfficientDet D4 by yourself on MSCOCO?

Your source image:
![source](https://user-images.githubusercontent.com/4096485/81574784-ebed2000-93ae-11ea-95bb-987eb534c730.jpg)


----

Due to problem of MSCOCO-labels as described in point (1), objectness of persons are very low in the crowds.

So if you run:
`./darknet detector test cfg/coco.data cfg/yolov4.cfg yolov4.weights -thresh 0.05 -ext_output source.jpg`

![predictions](https://user-images.githubusercontent.com/4096485/81574871-0626fe00-93af-11ea-8522-64d271709f91.jpg)

**Comment by WongKinYiu** [comment]
EfficientDet ignore bounding boxes with `iscrowd=1` tag in MSCOCO ground truth.
`annotations_ids = self.coco.getAnnIds(imgIds=self.image_ids[image_index], iscrowd=False)`
https://github.com/signatrix/efficientdet/blob/master/src/dataset.py#L66

**Comment by WongKinYiu** [comment]
EfficientDet ignore bounding boxes with `iscrowd=1` tag in MSCOCO ground truth when training.
`h.skip_crowd_during_training = True`
https://github.com/google/automl/blob/master/efficientdet/hparams_config.py#L186

**Comment by AlexeyAB** [comment]
What MSCOCO looks like for training Yolo:
![66252199-20905d80-e776-11e9-885f-73c4efc4751d](https://user-images.githubusercontent.com/4096485/81577870-b21e1880-93b2-11ea-96a3-916be31065ef.png)

**Comment by wwzh2015** [comment]
Thanks  everyone！

**Comment by zf921022** [comment]
>     1. YOLOv4 was trained on MSCOCO-2014 that is labeled with crowds=1, so the crowd of people is marked as one object in the MSCOCO-dataset, as you see in your yolov4-detection image. This can greatly reduce the detection accuracy of individual persons in a crowd. So YOLOv4 should be retrained with correct labels.
> 
>     2. For each neural network, you can find a picture with which one neural network is working poorly and the other is good, and vice versa. Therefore, for testing instead of 1 image there are used 20,000 images from the Microsoft COCO test-dev dataset
> 
>     3. EfficientDet D4 -  49.0% AP - **13 FPS** (Volta), while YOLOv4 - 43.5% AP - **62 FPS** (Volta), https://arxiv.org/pdf/1911.09070.pdf and https://arxiv.org/pdf/2004.10934.pdf so yes EfficientDet D4 is **+5.5% more accurate and 4.7x time slower** than YOLOv4
> 
>     4. EfficientDet D4 was trained on network 1024x1024 while YOLOv4 was trained on 512x512, so EfficientDet D4 can better detect small objects
> 
>     5. Did you ever try to train and test EfficientDet D4 by yourself on MSCOCO?
> 
> 
> Your source image:
> ![source](https://user-images.githubusercontent.com/4096485/81574784-ebed2000-93ae-11ea-95bb-987eb534c730.jpg)
> 
> Due to problem of MSCOCO-labels as described in point (1), objectness of persons are very low in the crowds.
> 
> So if you run:
> `./darknet detector test cfg/coco.data cfg/yolov4.cfg yolov4.weights -thresh 0.05 -ext_output source.jpg`
> 
> ![predictions](https://user-images.githubusercontent.com/4096485/81574871-0626fe00-93af-11ea-8522-64d271709f91.jpg)

absolutely right！
